tools/code_format/rst_check.py

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/python3
+
+import subprocess
+import sys
+
+# TODO(phlax): add rstcheck also
+
+# things we dont want to see in generated docs
+# TODO(phlax): move to .rstcheck.cfg when available
+RSTCHECK_GREP_FAIL = (" ref:", "\\[\\#")
+
+
+def run_grep_check(check):
+    command = ["grep", "-nr", "--include", "\\*.rst"] + [check]
```

Comment:
Is the plan to expand this beyond just grep checking? If it's a grep wrapper, then a shell script is fine.

Reply:
my plan is to rewrite it with the proper rstcheck (ie #15786 ) and using class like in #15802 so its kinda temporary to do it this way - but it fixes the immediate issue and introduces the generated rst checking